bpjs-debugger/src/main/java/il/ac/bgu/se/bp/execution/BPJsDebuggerRunnerImpl.java

```diff
@@ -294,34 +284,48 @@ private void removeExternalEvents(EventSelectionResult esr) {
         return resolveFuture(new Stop().applyCommand(debuggerEngineImpl));
     }
 
+    @Override
+    public FutureTask<String> getState() {
+
+        if (this.state.getDebuggerState() == RunnerState.State.JS_DEBUG)
+            return createResolvedFuture("Cant get state in JS_DEBUG. updated state will be send when receiving");
+        else if (this.state.getDebuggerState() == RunnerState.State.RUNNING)
+            return createResolvedFuture("BProg already in running state. you will get the state when reach the next sync state");
+        else if (!isStarted())
+            return createResolvedFuture("BProg not started yet");
+        logger.debug("Generate state from getState");
+        BPDebuggerState bpDebuggerState = debuggerEngineImpl.generateDebuggerState();
+        System.out.println(bpDebuggerState.toString());
+        return resolveFuture(new GetState().applyCommand(debuggerEngineImpl));
+    }
+
```

Comment:
we should keep the socket mechanism in this request as well.
when the client requests _getState_, we should send it using the socket, as here it is being resolved immediately as string.

we can keep it for now, but for future plan, i think it's better to use the socket from 1 method, and to rename this command for something like 'sendState' instead of get, which will only trigger and send the 'generateState' result using the socket.